test/e2e/validation_test.go

```diff
@@ -689,3 +685,22 @@ func (tc *testContext) validateUpgradeableCondition(expected meta.ConditionStatu
 	}
 	return nil
 }
+
+// getOperatorConditionName returns the operator condition name using the env var present in the deployment
+func (tc *testContext) getOperatorConditionName() (string, error) {
+	deployment, err := tc.client.K8s.AppsV1().Deployments(resourceNamespace).Get(context.TODO(), resourceName,
+		meta.GetOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "error getting operator deployment")
+	}
+	// Get the operator condition name using the deployment spec
+	for i := range deployment.Spec.Template.Spec.Containers {
+		envVars := deployment.Spec.Template.Spec.Containers[i].Env
```

Comment:
Do we know the name of the container beforehand, so that we can filter out the one we are not interested in? Otherwise,  this logic with loop all the `envVars` for all containers.

Reply:
that's correct, updating it now.